src/cmds/core/ban.py

```diff
@@ -24,6 +24,49 @@ class BanCog(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    class BanApprove(discord.ui.view):
+        def __init__(self, ban_id):
+            super().__init__(timeout=None)
+            self.ban_id = ban_id
+        @discord.ui.button(label="Approve", style=discord.ButtonStyle.primary)
+        async def approve_ban(self, interaction: discord.Interaction, button: discord.ui.Button):
+            async with AsyncSessionLocal() as session:
+                ban_to_update = await session.get(Ban, self.ban_id)
+                if not ban_to_update:
+                    message = "Cannot find record of ban request. Has this user already been unbanned?"
+
+                ban_to_update.approved = True
+                await session.commit()
+
+                stmt = select(Ban).filter(Ban.id == self.ban_id)
+                result = await session.scalars(stmt)
+                ban: Ban = result.first()
+
+            member = await self.bot.get_member_or_user(interaction.guild, ban.user_id)
```

Comment:
Here `self` refers to the instance of the class `BanApprove`, which doesn't have access to `bot` member that exists in "parent class" `BanCog`